service/init.go

```diff
@@ -534,3 +541,85 @@ func (amf *AMF) SendNFProfileUpdateToNrf() {
 		}
 	}
 }
+
+func UeConfigHandler(supi, sst, sd string, msg interface{}) {
+	amfSelf := context.AMF_Self()
+	ue, _ := amfSelf.AmfUeFindBySupi("imsi-" + supi)
+	if ue != nil && ue.State[models.AccessType__3_GPP_ACCESS].Is(context.Registered) {
+		var nwInitieadDetach bool
+		// Triggers for NwInitiatedDeRegistration
+		// - Only 1 Allowed Nssai is exist and its slice information matched
+		ns := msg.(*protos.NetworkSlice)
+		if len(ue.AllowedNssai[models.AccessType__3_GPP_ACCESS]) == 1 {
+			nwInitieadDetach = true
+		} else {
+
+			//TODO send configuration update to update allowed nssai list with re-registration required to UE
+			//     so that pdu session sync up happen in registration procedure when UE triggers this procedure
+			//gmm_message.SendConfigurationUpdateCommand(ue, models.AccessType__3_GPP_ACCESS, nil)
+
+			//we are doing local cleanup for now, below code will be deprecated when we support
+			// Configuration Update
+			sst, _ := strconv.Atoi(ns.Nssai.Sst)
+			ue.SmContextList.Range(func(key, value interface{}) bool {
+				smContext := value.(*context.SmContext)
+				if smContext.Snssai().Sst == int32(sst) && smContext.Snssai().Sd == ns.Nssai.Sd {
+					logger.CfgLog.Infof("Deleted Slice [sst: %v, sd: %v]matched with smcontext, sending Release SMContext Request to SMF",
+						smContext.Snssai().Sst, smContext.Snssai().Sd)
+					//send smcontext release request
+					problemDetail, err := consumer.SendReleaseSmContextRequest(ue, smContext, nil, "", nil)
+					if problemDetail != nil {
+						ue.GmmLog.Errorf("Release SmContext Failed Problem[%+v]", problemDetail)
+					} else if err != nil {
+						ue.GmmLog.Errorf("Release SmContext Error[%v]", err.Error())
+					}
+
+				}
+				return true
+			})
+
+			var allowedList []models.AllowedSnssai
+			//update Allowed Nssai List
+			for _, slice := range ue.AllowedNssai[models.AccessType__3_GPP_ACCESS] {
+				if slice.AllowedSnssai.Sst != int32(sst) && slice.AllowedSnssai.Sd != ns.Nssai.Sd {
+					allowedList = append(allowedList, slice)
+				}
+			}
+			ue.AllowedNssai[models.AccessType__3_GPP_ACCESS] = allowedList
+		}
+		// TODO:more than one slice, trigger UEConfigurationUpdate Procedure
+		if nwInitieadDetach {
+			gmm.GmmFSM.SendEvent(ue.State[models.AccessType__3_GPP_ACCESS], gmm.NwInitiatedDeregistrationEvent, fsm.ArgsType{
+				gmm.ArgAmfUe:      ue,
+				gmm.ArgAccessType: models.AccessType__3_GPP_ACCESS,
+			})
+		}
+	} else {
+		logger.CfgLog.Warnf("Ue is not in Registered State")
```

Comment:
probably we should add "nwInitieadDetach" handling for all states, like not registered condition here